flink-runtime/src/main/java/org/apache/flink/runtime/webmonitor/WebMonitorEndpoint.java

```diff
@@ -714,6 +717,7 @@ public void startInternal() throws Exception {
 	@Override
 	protected CompletableFuture<Void> shutDownInternal() {
 		executionGraphCache.close();
+		ExecutorUtils.gracefulShutdown(10, TimeUnit.SECONDS, executor);
```

Comment:
I think we should shut down the `executor` after shutting down the handlers. Otherwise we might run into the situation that a handler tries to add something to a closed `Executor`.